docs/src/main/asciidoc/virtual-threads.adoc

```diff
@@ -6,99 +6,166 @@ https://github.com/quarkusio/quarkus/tree/main/docs/src/main/asciidoc
 = Writing simpler reactive REST services with Quarkus Virtual Thread support
 
 include::_attributes.adoc[]
+:diataxis-type: reference
 :resteasy-reactive-api: https://javadoc.io/doc/io.quarkus.resteasy.reactive/resteasy-reactive/{quarkus-version}
-:resteasy-reactive-common-api: https://javadoc.io/doc/io.quarkus.resteasy.reactive/resteasy-reactive-common/{quarkus-version}
 :runonvthread: https://javadoc.io/doc/io.smallrye.common/smallrye-common-annotation/latest/io/smallrye/common/annotation/RunOnVirtualThread.html
 :blockingannotation: https://javadoc.io/doc/io.smallrye.common/smallrye-common-annotation/latest/io/smallrye/common/annotation/Blocking.html
 :vthreadjep: https://openjdk.org/jeps/425
 :thread: https://docs.oracle.com/en/java/javase/18/docs/api/java.base/java/lang/Thread.html
-:mutiny-vertx-sql: https://smallrye.io/smallrye-mutiny-vertx-bindings/2.26.0/apidocs/io/vertx/mutiny/sqlclient/package-summary.html
 :pgsql-driver: https://javadoc.io/doc/org.postgresql/postgresql/latest/index.html
 
-This guide explains how to benefit from Java 19 virtual threads when writing REST services in Quarkus.
+This guide explains how to benefit from Java 19+ virtual threads in Quarkus application.
 
-[TIP]
-====
-This is the reference guide for using virtual threads to write reactive REST services.
-Please refer to the xref:rest-json.adoc[Writing JSON REST services guides] for a lightweight introduction to reactive REST
-services and to the xref:resteasy-reactive.adoc[Writing REST Services with RESTEasy Reactive] guide for a detailed presentation.
-====
-
-== What are virtual threads ?
+== What are virtual threads?
 
 === Terminology
 OS thread::
-A "thread-like" data-structure managed by the Operating System.
+A "thread-like" data structure managed by the Operating System.
 
 Platform thread::
-Up until Java 19, every instance of the link:{thread}[Thread] class was a platform thread, that is, a wrapper around an OS thread.
-Creating a platform threads creates an OS thread, blocking a platform thread blocks an OS thread.
+Until Java 19, every instance of the link:{thread}[Thread] class was a platform thread, a wrapper around an OS thread.
+Creating a platform thread creates an OS thread, and blocking a platform thread, consequently, blocks an OS thread.
 
 Virtual thread::
```

Comment:
We might want to add a reference to https://dl.acm.org/doi/10.1145/3583678.3596895